webview-ui/src/context/ExtensionStateContext.tsx

```diff
@@ -140,7 +140,16 @@ export const ExtensionStateContextProvider: React.FC<{ children: React.ReactNode
 	}, [])
 
 	const handleInputChange = useCallback(
-		(field: keyof ApiConfiguration) => (event: any) => {
+		(field: keyof ApiConfiguration, softUpdate?: boolean) => (event: any) => {
+			if (softUpdate === true) {
```

Comment:
@samhvw8 do you mind explaining the issue and how this fixes it? Thank you!

Reply:
@mrubens Current flow is when update a field on api configuration
i call upsert to api config profile and build api handle
that flow is ok but sometimes build api handle occure error when change api provider field because missing key field
so i make an sort update when change api provider